scaleway/resource_object_bucket.go

```diff
@@ -197,12 +197,21 @@ func resourceScalewayObjectBucket() *schema.Resource {
 							Description: "Enable versioning. Once you version-enable a bucket, it can never return to an unversioned state",
 							Type:        schema.TypeBool,
 							Optional:    true,
-							Default:     false,
+							Computed:    true,
 						},
 					},
 				},
 			},
 		},
+		CustomizeDiff: func(ctx context.Context, diff *schema.ResourceDiff, meta interface{}) error {
+			if diff.Get("object_lock_enabled").(bool) {
+				if diff.HasChange("versioning") && !diff.Get("versioning.0.enabled").(bool) {
+					return fmt.Errorf("versioning must be enabled when object lock is enabled")
+				}
+			}
+
+			return nil
+		},
```

Comment:
What happened with this steps ?
```hcl
resource "scaleway_object_bucket" "backup" {
  versioning {
    enabled = false
  }
}
```
```hcl
resource "scaleway_object_bucket" "backup" {
  object_lock_enabled = true
  versioning {
    enabled = false
  }
}
```

Reply:
`object_lock_enabled` has `ForceNew` enabled so it will display the proper `versioning must be enabled when object lock is enabled` error during plan